### ISSUE TITLE
?/amp why are my amp pages like this 

### ISSUE THREAD

**Issue by jillianok** [Automattic/amp-wp#474]
I have paid a computer person to install plugin and setup amp on my website. However to see the amp pages you have do
mydomain.com.au/?amp.  not /amp.  Why is this. I have not seen any other amp pages with the question mark ?

**Comment by cayci** [comment]
The question mark marks the beginning of a query string.  Please refer to [a simple description of query strings and why they are used](https://launchschool.com/books/http/read/what_is_a_url#querystringsparameters)

From the [WordPress.org plugin directory's AMP plugin page:](https://wordpress.org/plugins/accelerated-mobile-pages/)
"How can I view the AMP version of my site?"
"To view the AMP version, add /?amp at the end of your url.  
An example would be http://Website.com/?amp"

**Comment by DrLightman** [comment]
With permalinks enabled you should have /amp, not "amp" as a GET parameter

**Comment by amedina** [comment]
Make sure that permalinks are enabled in your site. And thanks for using the plugin :)

**Comment by emresque** [comment]
Hi,

I still get the question mark although i have permalinks enabled and my normal pages and posts have urls without question marks. And when i just type /amp et the end of my urls it goes 404. Can anybody help me on this? (example page-->[seo hizmeti](https://www.netseouzmani.com/seo-hizmeti/?amp)

**Comment by westonruter** [comment]
@emresque The query var is used for pages and any other non-hierarchical post type. Please see https://wordpress.org/support/topic/0-6-0-not-working-for-pages/#post-9903766

**Comment by emresque** [comment]
ah now i see, it is true that for blog posts i have normal urls without the question mark..good to know that it is ok from google's point of view..thanks for the information Weston!

**Comment by CVANL** [comment]
Hi Folks, I started using AMP at my wordpress website 4 days ago. In the first 2 days my amp pages were detected with /amp/ at the end. But since yesterday it has been changed to ?amp

The problem is that website is now redirecting the browser at pages from /amp/ to the desktop url without amp. this is an example of a page:

1. https://coronavirusactueel.nl/corona-kaart-nederland/amp/

**_When you visit this url, the amp is not showing up. But there is an AMP version 100% sure._**

2. https://coronavirusactueel.nl/burgemeesters-bespreken-overlast-motorrijders/amp/

**_This is an example if a post and when you visit this url it's redirected to ?amp and that's an AMP version of the post._**

Could it be that my caching plugin wp-rocket or the google mod_pagespeed causes this problem? Because I started using them since yesterday.

I am afraid that this will mess up my google indexing and that I will have duplicate pages for more that 2000 posts. 

When you search in google: **site:coronavirusactueel.nl amp** you can see that /amp/ is what google indexed. But these pages don't work anymore. 

Could somebody advice me what to do? 

Thanks! And greetings from the Netherlands.

Ray

**Comment by westonruter** [comment]
@CVANL I presume you used a different AMP plugin previously?

 If you look at the source code of https://coronavirusactueel.nl/corona-kaart-nederland/ you'll see:

```html
<link rel="amphtml" href="https://coronavirusactueel.nl/corona-kaart-nederland/?amp">
```

So the AMP version of a page (or another hierarchical post type) does not end in `/amp/` but `?amp`.

Nevertheless a post like https://coronavirusactueel.nl/burgemeesters-bespreken-overlast-motorrijders/ has:

```html
<link rel="amphtml" href="https://coronavirusactueel.nl/burgemeesters-bespreken-overlast-motorrijders/?amp">
```

So for the AMP version of that post ends in `?amp`.

Note that the inconsistency of URLs for posts and pages will be addressed in #2204 where we'll default to `?amp` for all URLs.

Google's index will update to to use the new AMP URLs. 

Nevertheless, I'm not sure why attempting to access https://coronavirusactueel.nl/corona-kaart-nederland/amp/ is redirecting to https://coronavirusactueel.nl/corona-kaart-nederland/ and not https://coronavirusactueel.nl/corona-kaart-nederland/?amp

It's most likely because the AMP plugin doesn't even try to add the `/amp/` endpoint to pages:

https://github.com/ampproject/amp-wp/blob/1d641245c5e14af980596c4da1c2a4c0863fb4ee/includes/amp-helper-functions.php#L100

Because of this, WordPress's own canonical redirects are kicking in.

What you could in the short term while waiting for Google to re-index your pages is add a plugin that does this:

```php
namespace AMP_Register_Page_Endpoint;

function add_amp_endpoint() {
	\add_rewrite_endpoint( amp_get_slug(), EP_PERMALINK | EP_PAGES );
}

register_activation_hook( __FILE__, function () {
	flush_rewrite_rules( false );
	add_amp_endpoint();
} );

add_action( 'init', function() {
	add_amp_endpoint();
} );
```

I've added this to an existing extension plugin that deals with other aspects of AMP endpoint migration which you can use to do this: https://gist.github.com/westonruter/6b7d3b168711ee03c4ee29b68c3719b2

I've added a link to installation instructions in the comments.

**Comment by DrLightman** [comment]
Try visiting Settings | Permalinks page of your dashboard to force a flush of rewrite rules.
